gnd/src/main/java/com/google/android/gnd/rx/Result.java

```diff
@@ -62,12 +63,32 @@ public Throwable getError() {
     return error;
   }
 
-  public static <T, R> Function<T, Single<Result<R>>> wrapErrors(
+  public static <T, R> Function<T, Single<Result<R>>> wrapSingle(
       @NonNull Function<T, Single<R>> fn) {
     return (T value) -> fn.apply(value).map(Result::success).onErrorReturn(Result::error);
   }
 
-  public static <T> Consumer<? super Result<T>> unwrapErrors(
+  public static <T, R> Function<T, Observable<Result<R>>> wrapObservable(
+      @NonNull Function<T, Observable<R>> fn) {
+    return (T value) -> fn.apply(value).map(Result::success).onErrorReturn(Result::error);
+  }
+
+  /**
+   * Wraps observables in a Result without applying any operation. Equivalent to mapping "id" on the
+   * observable or "casting" or "lifting" a given observable into a Result.
```

Comment:
Casting has a specific meaning in Java, never heard of lifting - is that a term? Would "wrap" be more accurate?

Reply:
I ported over a bunch of terminology that isn't really common or applicable in Java :sweat: --sorry just a result of my natural thought process and not double checking myself.

I do think "wrap" is probably the most accurate term for what I'm trying to say here, since, as you note, 'cast' has a rather particular meaning. Basically, I just mean to convey that we "inject" the given Observable into a Result, that we take an `Observable<T>` type and produce a `Result<Observable<T>>` without performing any transformations on the `Observable<T>` in the process. I'll give this doc another go.

Actually, my use of the term "lifting" here is not even correct from the perspective I was writing from -.-, since one lifts functions, not types or values. Basically, `lift id<T>` makes id well defined for whatever type defines `lift` (so, if you have a list of integers and `id` defined on integer, `lift id` allows you to successfully run id on each element of the list 'lifting' the id function into the list context). For lists, lift is the same as `map` `lift id = map id`. There are other variants of lift that allow you to lift n-ary functions, such as a binary function like `+`--the convenience here is the elimination of lambdas `lift2 + = map (x -> (y -> x + y))`. If you're curious, the term comes from the FP realm: https://wiki.haskell.org/Lifting.